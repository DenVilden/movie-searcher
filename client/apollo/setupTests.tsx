import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from '../pages/_app';

// eslint-disable-next-line jest/prefer-spy-on
HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) =>
  render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  );

export * from '@testing-library/react';
