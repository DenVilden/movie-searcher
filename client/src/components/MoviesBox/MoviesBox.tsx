import { Grid, Grow, Typography } from "@material-ui/core";
import { MovieCard } from "..";
import { Root, Wrapper } from "./MoviesBox.styles";

type Props = {
  movies: {
    id: number;
    title: string;
    vote_average?: number;
    poster_path?: string | null;
    release_date?: string | null;
  }[];
  title: string;
  elevation?: number;
  padding?: number;
};

const MoviesBox = ({ movies, title, elevation = 10, padding = 1 }: Props) => (
  <Grow in>
    <Root elevation={elevation} padding={padding}>
      {!!title && (
        <Typography align="center" gutterBottom variant="h4">
          {title}
        </Typography>
      )}
      <Grid container>
        {movies.map((movie) => (
          <Wrapper
            key={movie.id}
            container
            data-testid="search-result"
            item
            justify="center"
            md={3}
            xs={6}
          >
            <MovieCard movie={movie} />
          </Wrapper>
        ))}
      </Grid>
    </Root>
  </Grow>
);

export default MoviesBox;
