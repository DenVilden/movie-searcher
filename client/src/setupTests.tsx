import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from './pages/_app';
import defaultResolvers from './graphql/resolvers';

window.HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
  cache?: any;
  resolvers?: any;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks, addTypeName, resolvers, cache }: RenderApolloOptions = {}
) =>
  render(
    <MockedProvider
      addTypename={addTypeName}
      cache={cache}
      mocks={mocks}
      resolvers={resolvers || defaultResolvers}
    >
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>
  );

export * from '@testing-library/react';
