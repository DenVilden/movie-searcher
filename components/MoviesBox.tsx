import { Grid, Typography, Paper } from '@material-ui/core';
import styled from '@emotion/styled';
import { css } from '@emotion/react';
import MovieCard from './MovieCard';

interface RootProps {
  padding: number;
}

const Root = styled(Paper)<RootProps>`
  background: none;
  background-color: inherit;

  ${({ theme, padding }) =>
    padding &&
    css`
      margin: ${theme.spacing(3, 1, 3, 1)};
      padding: ${theme.spacing(2, 0, 4, 0)};
    `};
`;

const Wrapper = styled(Grid)`
  margin-top: 10px;

  ${({ theme }) => theme.breakpoints.up('sm')} {
    padding: ${({ theme }) => theme.spacing(0, 2, 0, 2)};
  }
`;

interface Props {
  movies: {
    id: number;
    title: string;
    vote_average?: number;
    poster_path?: string | null;
    release_date?: string | null;
    media_type?: string;
  }[];
  title: string;
  elevation?: number;
  padding?: boolean;
}

export default function MoviesBox({
  movies,
  title,
  elevation = 10,
  padding = true,
}: Props) {
  return (
    <Root elevation={elevation} padding={+padding}>
      {!!title && (
        <Typography align="center" gutterBottom variant="h4">
          {title}
        </Typography>
      )}
      <Grid container>
        {movies.map(movie => (
          <Wrapper
            key={movie.id}
            container
            item
            justifyContent="center"
            md={3}
            xs={6}
          >
            <MovieCard movie={movie} />
          </Wrapper>
        ))}
      </Grid>
    </Root>
  );
}
