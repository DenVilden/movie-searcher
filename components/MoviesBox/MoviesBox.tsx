import { Grid, Grow, Typography, Paper, PaperProps } from '@material-ui/core';
import styled from 'styled-components';
import { MovieCard } from '..';

interface RootProps extends PaperProps {
  padding: number;
}

const Root = styled((props: RootProps) => <Paper {...props} />)`
  background: none;
  background-color: inherit;
  margin: ${(props) => props.padding && props.theme.spacing(3, 1, 3, 1)};
  padding: ${(props) => props.padding && props.theme.spacing(2, 0, 4, 0)};
`;

const Wrapper = styled(Grid)`
  margin-top: 10px;

  ${(props) => props.theme.breakpoints.up('sm')} {
    padding: ${(props) => props.theme.spacing(0, 2, 0, 2)};
  }
`;

interface Props {
  movies: {
    id: number;
    title: string;
    vote_average?: number;
    poster_path?: string | null;
    release_date?: string | null;
  }[];
  title: string;
  elevation?: number;
  padding?: number;
}

export default function MoviesBox({
  movies,
  title,
  elevation = 10,
  padding = 1,
}: Props) {
  return (
    <Grow in>
      <Root elevation={elevation} padding={padding}>
        {Boolean(title) && (
          <Typography align="center" gutterBottom variant="h4">
            {title}
          </Typography>
        )}
        <Grid container>
          {movies.map((movie) => (
            <Wrapper
              key={movie.id}
              container
              data-testid="search-result"
              item
              justify="center"
              md={3}
              xs={6}
            >
              <MovieCard movie={movie} />
            </Wrapper>
          ))}
        </Grid>
      </Root>
    </Grow>
  );
}
