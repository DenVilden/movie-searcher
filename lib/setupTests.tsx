/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/preact'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import '@testing-library/jest-dom/extend-expect'

const theme = createMuiTheme()

type RenderApolloOptions = {
  addTypeName?: boolean
  mocks?: MockedResponse[]
}

export function renderApollo(
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) {
  return render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  )
}

export { screen, fireEvent } from '@testing-library/preact'
