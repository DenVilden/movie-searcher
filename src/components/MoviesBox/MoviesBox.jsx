import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Grow, Typography } from '@material-ui/core';
import { Root, List, Wrapper } from './MoviesBox.styles';
import MovieCard from '../MovieCard/MovieCard';

const propTypes = {
  movies: PropTypes.arrayOf(PropTypes.any).isRequired,
  title: PropTypes.string,
  elevation: PropTypes.number,
  padding: PropTypes.number,
  clearInputValue: PropTypes.func,
};

const defaultProps = {
  title: '',
  elevation: 10,
  padding: 1,
  clearInputValue: null,
};

const MoviesBox = ({ movies, title, elevation, padding, clearInputValue }) => {
  if (!movies.length) return null;

  return (
    <Root elevation={elevation} padding={padding}>
      {title && (
        <Typography align="center" gutterBottom variant="h4">
          {title}
        </Typography>
      )}
      <List container>
        {movies.map(movie => (
          <Grow key={movie.id} in>
            <Grid item lg={2} md={3} sm={4} xs={6}>
              <Wrapper container justify="center">
                <Grid item>
                  <MovieCard clearInputValue={clearInputValue} movie={movie} />
                </Grid>
              </Wrapper>
            </Grid>
          </Grow>
        ))}
      </List>
    </Root>
  );
};

MoviesBox.propTypes = propTypes;
MoviesBox.defaultProps = defaultProps;

export default MoviesBox;
