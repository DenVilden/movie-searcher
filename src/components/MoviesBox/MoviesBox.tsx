/* eslint-disable camelcase */
import React from 'react';
import { Grid, Grow, Typography } from '@material-ui/core';
import { Root, List, Wrapper } from './MoviesBox.styles';
import MovieCard from '../MovieCard/MovieCard';

const defaultProps = {
  title: '',
  elevation: 10,
  padding: 1,
};

type Props = {
  movies: {
    id: number;
    title: string;
    vote_average?: number;
    poster_path: string | null;
    release_date?: string;
  }[];
  clearInputValue?: () => void;
} & typeof defaultProps;

const MoviesBox = ({
  movies,
  title,
  elevation,
  padding,
  clearInputValue,
}: Props) => (
  <Root elevation={elevation} padding={padding}>
    {title && (
      <Typography align="center" gutterBottom variant="h4">
        {title}
      </Typography>
    )}
    <List container>
      {movies.map(movie => (
        <Grow key={movie.id} in>
          <Grid item lg={2} md={3} sm={4} xs={6}>
            <Wrapper container justify="center">
              <Grid item>
                <MovieCard clearInputValue={clearInputValue} movie={movie} />
              </Grid>
            </Wrapper>
          </Grid>
        </Grow>
      ))}
    </List>
  </Root>
);

MoviesBox.defaultProps = defaultProps;

export default MoviesBox;
