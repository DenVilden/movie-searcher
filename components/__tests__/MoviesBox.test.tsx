import { renderApollo } from '../../lib/setupTests';
import MoviesBox from '../MoviesBox';

const mock = [
  {
    id: 1,
    poster_path: null,
    release_date: '2020',
    title: 'test',
    vote_average: 5,
  },
];

describe('moviesBox', () => {
  it('should take a snapshot', () => {
    const { asFragment } = renderApollo(
      <MoviesBox movies={mock} title="new" />,
    );

    const element = asFragment();

    expect(element).toMatchSnapshot();
  });
});
