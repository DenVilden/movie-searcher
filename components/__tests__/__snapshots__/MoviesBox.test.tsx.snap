// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`moviesBox should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation10 css-exwu77-MuiPaper-root"
    padding="1"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter MuiTypography-gutterBottom css-1trk376-MuiTypography-root"
    >
      new
    </h4>
    <div
      class="MuiGrid-root MuiGrid-container css-15fev7r-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-nks5yq-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation10 MuiCard-root css-1jin0gs-MuiPaper-root-MuiCard-root"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                />
              </div>
              <img
                alt="test"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
            <div
              class="MuiCardContent-root css-2zy926-MuiCardContent-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
              >
                test
              </h6>
              <p
                class="MuiTypography-root MuiTypography-body1 css-i0o70y-MuiTypography-root"
              >
                <span
                  class="css-1rbcxmo"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                   5
                </span>
              </p>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
